src/frequenz/client/electricity_trading/_client.py

```diff
@@ -83,12 +83,15 @@ def validate_decimal_places(value: Decimal, decimal_places: int, name: str) -> N
 class Client(BaseApiClient[ElectricityTradingServiceStub, grpc.aio.Channel]):
     """Electricity trading client."""
 
-    def __init__(self, server_url: str, connect: bool = True) -> None:
+    def __init__(
+        self, server_url: str, connect: bool = True, key: str | None = None
+    ) -> None:
         """Initialize the client.
 
         Args:
             server_url: The URL of the Electricity Trading service.
             connect: Whether to connect to the server immediately.
+            key: The API key for the authorization.
```

Comment:
Optional suggestion: The name `key` can be more descriptive for the intended use. I'd rename it to `auth_key`, `api_key`, or something around this